url9d32402837_dusan_rychnovsky_chess_endgames_tgz-pJ8-cz_dusanrychnovsky_chessendgames_core_strategies_PrecomputedValuesJ8/src/cz/dusanrychnovsky/chessendgames/gui/GUIAdapter.java

```diff
@@ -0,0 +1,371 @@
+package cz.dusanrychnovsky.chessendgames.gui;
+
+import java.io.BufferedInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.ObjectInputStream;
+
+import cz.dusanrychnovsky.chessendgames.core.Game;
+import cz.dusanrychnovsky.chessendgames.core.King;
+import cz.dusanrychnovsky.chessendgames.core.Move;
+import cz.dusanrychnovsky.chessendgames.core.Player;
+import cz.dusanrychnovsky.chessendgames.core.Position;
+import cz.dusanrychnovsky.chessendgames.core.Rook;
+import cz.dusanrychnovsky.chessendgames.core.Situation;
+import cz.dusanrychnovsky.chessendgames.core.Player.Color;
+import cz.dusanrychnovsky.chessendgames.core.strategies.PrecomputedValues;
+import cz.dusanrychnovsky.chessendgames.core.strategies.Strategy;
+
+/**
+ * 
+ * @author Dušan Rychnovský
+ *
+ */
+public class GUIAdapter implements MouseEventListener
+{
+	private final Window window;
+	
+	private final Player whitePlayer = Player.get(Color.WHITE);
+	private final King whiteKing = new King(whitePlayer);
+	
+	private final Player blackPlayer = Player.get(Color.BLACK);
+	private final King blackKing = new King(blackPlayer);
+	private final Rook blackRook = new Rook(blackPlayer);
+	
+	private final Strategy strategy;
+	private Game game;
+	
+	private GameState currentState;
+	
+	/**
+	 * 
+	 * @param args
+	 * @throws ClassNotFoundException
+	 * @throws IOException
+	 */
+	public static void main(String[] args) throws ClassNotFoundException, IOException
+	{
+		new GUIAdapter();
+	}
+	
+	/**
+	 * 
+	 * @param strategy
+	 * @throws IOException 
+	 * @throws ClassNotFoundException 
+	 */
+	public GUIAdapter() throws ClassNotFoundException, IOException 
+	{
+		SplashScreen splashScreen = new SplashScreen();
+		
+		InputStream is = PrecomputedValues.class.getResourceAsStream("strategy.dat");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *is*. The resource is referenced at lines 62, 64. The resource is closed at line 65. There are other execution paths that do not contain closure statements, for example, when PrecomputedValues.load throws an exception. Ensure *is* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).